### PR TITLE
Fix bugs around Global and Detailed search

### PR DESCRIPTION
**Pull request by sri2k1us**
- Fix intercom logging so that it logs only when search results are updated.
- Fix copying DE link from details page.
- Show search results from `Community Data` and `Shared with Me` in Detailed search results page.
- Fix `Copy DE Link` tool tip.

### CODE REVIEW

File: src/components/search/detailed/DataSearchResults.js [hunk 1]
```diff
@@ -149,21 +143,25 @@ function DataSearchResults(props) {
...
+                    userHomeDir: "",
                     rowsPerPage: searchConstants.DETAILED_SEARCH_PAGE_SIZE,
                     sortField: sortField,
                     sortDir: sortOrder,
                 },
             ]);
             setDataSearchQueryEnabled(true);
         }
-    }, [searchTerm, sortField, sortOrder, userHomeDir]);
+    }, [searchTerm, sortField, sortOrder]);
```
Comment (sri2k1us) [inline src/components/search/detailed/DataSearchResults.js:154]: Now I think about it - this still doesn't cover 1 use case where there are no search results matching the the term entered by users. In that case we don't show the `See all results for `term` option in the drop-down. So user has to hit `enter` to see all the results. May be we should that option even when there are no matching top 10 results for `Data` 🤔

Comment (psarando) [reply src/components/search/detailed/DataSearchResults.js:154]: Sounds like a good idea to me 👍

Comment (sri2k1us) [reply src/components/search/detailed/DataSearchResults.js:154]: Ok this done.

**Review by ash3rz — APPROVED** [review]
LGTM :shipit:

**Review by psarando — APPROVED** [review]
LGTM :shipit: 👍

**Comment by sri2k1us** [comment]
Thanks for the reviews.

**Pull request merged**